### DISCUSSION TITLE
Switch from DirichletBC to NeumannBC

### DISCUSSION THREAD

**Discussion by styyokuda** [idaholab/moose#24596] [Q&A General] [answered]
All,

My model starts with a Dirichlet boundary condition with a zero value but when the flux value reaches a specified value
I would like to switch a Dirichlet boundary to a Neumann boundary condition with a flux value calculated in a Material object.
I know "TimePeriod" in [Control] is available but I have not found a way to activate the boundary condition switch by a parameter available in my model.
Please help. S. Thomas

**Comment by GiudGiud** [comment, marked as answer]
Hello

Control is the way to go.
What error are you getting?

Guillaume

**Reply by styyokuda** [reply]
Thank you, Guillaume.
I have not figured it out yet. I can make a switch by specifying a time using "TimePeriod" but I would like to do the switch when the flux value reaches a specified value calculated in a Material object. I still don't know how to do that.
S. Thomas

**Reply by GiudGiud** [reply]
Use this one instead of TimePeriod:
https://mooseframework.inl.gov/source/controls/ConditionalFunctionEnableControl.html

the function can depend on postprocessors (ParsedFunction) and the PP can look a the flux

**Reply by styyokuda** [reply]
Guillaume,
I looked at the examples in the link. They are hard to understand.  Do you have friendlier ones?
Please help. S. Thomas

**Reply by GiudGiud** [reply]
can you paste your BCs block here?
and how do you compute the criterion for the switch? is it in your postprocessors already?


 I ll write the control for you

**Reply by styyokuda** [reply]
[Control.zip](https://github.com/idaholab/moose/files/11667942/Control.zip)

Guillaume,
It would be appreciated if you could take a look at what I am trying to do.
In the attached files, FluxMaterial starts to calculate the flux with zero DirichletBC. The maximum flux is specified to be 150 and once the flux reaches 150, flux of 150 is set in MaterialNeumannBC. The challenge is how to switch from DirichletBC to MaterialNeumannBC for variable "u" in Block 0. I am using "TimePeriod" here but it is not accurate. Please help. I would like to thank you for your time and consideration.

**Reply by GiudGiud** [reply]
Try something like:

```
[Postprocessors]
    [max_flux]
    type = ElementExtremeMaterialProperty
    mat_prop = "flux_value"
    value_type = max
  []
[]

[Functions]
  [conditional_function]
    type = ParsedFunction'
    expression = 'if(max_flux > 150, 1, 0)'
    symbol_names = max_flux
    symbol_values = max_flux
  []
[]

[Controls]
    [switch_up]
    type = ConditionalFunctionEnableControl
    enable_objects = 'BCs::u_interface01_Dirichlet'
    disable_objects = 'BCs::u_interface01_Neumann'
    conditional_function = 'conditional_function'
  []
[]
